java/algorithm/src/main/java/org/apache/arrow/algorithm/search/UnSortedVectorSearcher.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.algorithm.search;
+
+import org.apache.arrow.algorithm.sort.VectorValueComparator;
+import org.apache.arrow.vector.ValueVector;
+
+/**
+ * Searcher for un-sorted vectors.
+ * @param <V> vector type.
+ */
+public class UnSortedVectorSearcher<V extends ValueVector> extends VectorSearcher<V> {
+
+  /**
+   * Constructor for un-sorted vector searcher.
+   *
+   * @param targetVector the vector to search for.
+   * @param comparator   the criteria to determine vector data equality.
+   */
+  public UnSortedVectorSearcher(V targetVector, VectorValueComparator<V> comparator) {
```

Comment:
```suggestion
  public UnsortedVectorSearcher(V targetVector, VectorValueComparator<V> comparator) {
```

Reply:
I have refactored the class name to LinearVectorSearcher, because it is a linear search algorithm.